### PR TITLE
Improve comment on `unit_interval` in cddl

### PR DESCRIPTION
**Pull request by lehins**
# Description

* Remove unused `rational`
* Add a better comment about `unit_interval`

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [ ] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

File: eras/conway/impl/cddl-files/extra.cddl [hunk 1]
```diff
@@ -11,9 +11,20 @@ nonempty_oset<a> = #6.258([+ a]) / [+ a]
 positive_int = 1 .. 18446744073709551615
 
 unit_interval = #6.30([1, 2])
-  ; real unit_interval is: #6.30([uint, uint])
-  ; but this produces numbers outside the unit interval
-  ; and can also produce a zero in the denominator
+  ; unit_interval = #6.30([uint, uint])
+  ;
+  ; Comment above depicts the acual definition for `unit_interval`.
```
Comment (teodanciu) [inline eras/conway/impl/cddl-files/extra.cddl]: `actual`

File: eras/conway/impl/cddl-files/extra.cddl [hunk 2]
```diff
@@ -11,11 +11,20 @@ nonempty_oset<a> = #6.258([+ a]) / [+ a]
 positive_int = 1 .. 18446744073709551615
 
 unit_interval = #6.30([1, 2])
```
Comment (Soupstraw) [inline eras/conway/impl/cddl-files/extra.cddl:13]: I'm thinking that it might be a good idea to have `#6.30([1, 1])` and  `#6.30([0, 1])` also as possibilities, that way we can test more edge cases.

**Review by Soupstraw — APPROVED** [review]
LGTM

**Review by teodanciu — APPROVED** [review]
Looks good to me

File: eras/conway/impl/cddl-files/extra.cddl (hunk 2 above)
Comment (lehins) [reply eras/conway/impl/cddl-files/extra.cddl:13]: It will be better to create a PR that splits out the concerns of testing and specifying.

One that will be used for testing will contain a whole bunch of random values, while the one o that used for specifying will have the correct types.

We can do it at some later point. I only created this PR because we were asked to clarify the comment on the `unit_interval`

File: eras/conway/impl/cddl-files/extra.cddl (hunk 1 above)
Comment (lehins) [reply eras/conway/impl/cddl-files/extra.cddl]: Done

File: eras/conway/impl/cddl-files/extra.cddl (hunk 2 above)
Comment (Soupstraw) [reply eras/conway/impl/cddl-files/extra.cddl:13]: I think it would be a good idea to extend CDDL so that it's possible to express these kinds of constraints. Maybe once `Cuddle` is in a more developed state we can think about that.

Comment (Soupstraw) [reply eras/conway/impl/cddl-files/extra.cddl:13]: I think [Cue](https://cuelang.org/) has some interesting ideas that we could borrow.

Comment (lehins) [reply eras/conway/impl/cddl-files/extra.cddl:13]: I created a ticket that captures this as a future task https://github.com/input-output-hk/cardano-ledger/issues/3931
